[docs] Start Dashboard Layout template based on Toolpad Core

apedroferreira opened the pull request:
Work in progress.

DiegoAndai commented:
Sorry for the delay.

The file structure looks good. I see some inconsistencies with the other templates, for example:

### Buttons

Buttons sizes are 40x40 on this one and 36x36 on the other ones. Also the icons inside them look too big. 

**CRUD dashboard**
<img width="67" alt="Screenshot 2025-07-08 at 14 12 29" src="https://github.com/user-attachments/assets/64382b7b-0c97-47de-b9da-e5e5be796ca6" />

**Dashboard**
<img width="60" alt="Screenshot 2025-07-08 at 14 12 25" src="https://github.com/user-attachments/assets/4c9c2250-c000-4946-82fe-f53a6b86a368" />

### Breadcrumbs

**CRUD dashboard**
<img width="266" alt="Screenshot 2025-07-08 at 14 11 53" src="https://github.com/user-attachments/assets/71c95370-5b8f-4eeb-983c-111dc8f93a93" />

**Dashboard**
<img width="167" alt="Screenshot 2025-07-08 at 14 11 57" src="https://github.com/user-attachments/assets/83bfe273-a0c8-4bb2-b6e3-e36299bf242a" />

###  Nav bar
**CRUD dashboard**
<img width="313" alt="Screenshot 2025-07-08 at 14 12 14" src="https://github.com/user-attachments/assets/3a11bb46-0ed2-49df-846d-42d56259ab66" />

**Dashboard**
<img width="222" alt="Screenshot 2025-07-08 at 14 12 08" src="https://github.com/user-attachments/assets/01d6f791-6576-4993-9f71-056eb106c992" />

I think we could copy a lot of these from the other templates so we can keep them consistent and of a higher quality.

Let me know if this is out of scope.

apedroferreira commented:
> With those four things fixes this looks good to me 👍🏼

I've addressed all your points, please take a look when you have the time and let me know if it's good now!

Janpot commented on docs/data/material/getting-started/templates/crud-dashboard/components/EmployeeEdit.js:
```diff
@@ -0,0 +1,191 @@
+      onFieldChange={handleFormFieldChange}
+      onSubmit={handleFormSubmit}
+      onReset={handleFormReset}
+      submitButtonLabel="Edit"
```
"Edit" feels to me like it's going to bring me to a page to edit the record, but I'm already there

```suggestion
      submitButtonLabel="Save"
```

Janpot commented on docs/data/material/getting-started/templates/crud-dashboard/data/employees.ts:
```diff
@@ -0,0 +1,201 @@
+    id: 1,
+    name: 'Edward Perry',
+    age: 25,
+    joinDate: new Date().toISOString(),
```
Looks like this caused [unstable screenshots](https://app.argos-ci.com/mui/material-ui/builds/39978/178329167)

oliviertassinari commented:
I'm late to this. Great idea as a temporary solution for Toolpad users 👍
